src/gromacs/nblib/topology.cpp

```diff
@@ -235,6 +259,11 @@ TopologyBuilder& TopologyBuilder::addMolecule(const Molecule& molecule, const in
     return *this;
 }
 
+void TopologyBuilder::addParticleTypesInteractions(ParticleTypesInteractions& particleTypesInteractions)
+{
+    particleTypesInteractions_ = std::make_shared<ParticleTypesInteractions>(particleTypesInteractions);
```

Comment:
Why does it need to be shared?

Reply:
I think this can indeed be a unique pointer